packages/runtime/src/useCases/transport/identityRecoveryKits/CreateIdentityRecoveryKit.ts

```diff
@@ -0,0 +1,72 @@
+import { Result } from "@js-soft/ts-utils";
+import { CoreDate } from "@nmshd/core-types";
+import { AccountController, Device, DevicesController, PasswordProtectionCreationParameters, TokenContentDeviceSharedSecret, TokenController } from "@nmshd/transport";
+import { Inject } from "@nmshd/typescript-ioc";
+import { TokenDTO } from "../../../types";
+import { RuntimeErrors, SchemaRepository, TokenAndTemplateCreationValidator, UseCase } from "../../common";
+import { TokenMapper } from "../tokens/TokenMapper";
+
+export interface CreateIdentityRecoveryKitRequest {
+    profileName: string;
```

Comment:
Why is `profileName` not optional?

Reply:
B/C the app heavily relies on that.

Reply:
Is this a name that you can freely assign, or do you have to refer to an existing profile name? If this is the case, you would need somewhere to validate that the profile name exists. Would this then take place in the App because we don't know the existing profile names?

Reply:
profile names are an app runtime thing so we cannot do that